src/tools/rbd/action/Nbd.cc

```diff
@@ -122,6 +122,11 @@ int execute_attach(const po::variables_map &vm,
     return -EINVAL;
   }
 
+  if (vm.count("snap-id")) {
+    args.push_back("--snap-id");
+    args.push_back(vm["snap-id"].as<std::string>());
```

Comment:
Here and elsewhere: let's add a "trying to access snapshot using both name and id" check.

Reply:
Sure. But unmap and detach commands expect either snapid or snapname as per what was actually placed at the map and attach commands respectively.